src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1348,126 +1051,23 @@ namespace winrt::Microsoft::Terminal::Control::implementation
-                {
-                    // Reset the selection
-                    _terminal->ClearSelection();
-                    _selectionNeedsToBeCopied = false; // there's no selection, so there's nothing to update
-                }
-
-                if (shiftEnabled && _terminal->IsSelectionActive())
-                {
-                    // If shift is pressed and there is a selection we extend it using the selection mode
-                    // (expand the "end"selection point)
-                    _terminal->SetSelectionEnd(terminalPosition, mode);
-                    _selectionNeedsToBeCopied = true;
-                }
-                else if (mode != ::Terminal::SelectionExpansionMode::Cell || shiftEnabled)
-                {
-                    // If we are handling a double / triple-click or shift+single click
-                    // we establish selection using the selected mode
-                    // (expand both "start" and "end" selection points)
-                    _terminal->MultiClickSelection(terminalPosition, mode);
-                    _selectionNeedsToBeCopied = true;
-                }
-
-                if (_terminal->IsSelectionActive())
-                {
-                    // GH#9787: if selection is active we don't want to track the touchdown position
-                    // so that dragging the mouse will extend the selection rather than starting the new one
-                    _singleClickTouchdownPos = std::nullopt;
-                }
-
-                _renderer->TriggerSelection();
-            }
-            else if (point.Properties().IsRightButtonPressed())
-            {
-                if (_settings.CopyOnSelect() || !_terminal->IsSelectionActive())
-                {
-                    // CopyOnSelect right click always pastes
-                    PasteTextFromClipboard();
-                }
-                else
-                {
-                    CopySelectionToClipboard(shiftEnabled, nullptr);
-                }
-            }
+            const auto contactRect = point.Properties().ContactRect();
+            auto anchor = winrt::Windows::Foundation::Point{ contactRect.X, contactRect.Y };
+            _interactivity->Touched(anchor);
         }
-        else if (ptr.PointerDeviceType() == Windows::Devices::Input::PointerDeviceType::Touch)
+        else
         {
-            const auto contactRect = point.Properties().ContactRect();
-            // Set our touch rect, to start a pan.
-            _touchAnchor = winrt::Windows::Foundation::Point{ contactRect.X, contactRect.Y };
+            _interactivity->PointerPressed(cursorPosition,
+                                           TermControl::GetPressedMouseButtons(point),
+                                           TermControl::PointerToMouseButtons(point),
+                                           point.Timestamp(),
+                                           ControlKeyStates(args.KeyModifiers()),

```

Comment:
If this is a constructor call, use {}.

```suggestion
                                           ControlKeyStates{ args.KeyModifiers() },
```

Reply:
(in a list of function calls it stands out better too!)